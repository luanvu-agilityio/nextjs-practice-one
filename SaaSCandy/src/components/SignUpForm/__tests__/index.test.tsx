import { render, screen } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import type { ReactNode } from 'react';
import { SignUpPageContent } from '../index';

// Mock Next.js
jest.mock('next/navigation', () => ({
  usePathname: () => '/signup',
  useRouter: () => ({
    push: jest.fn(),
  }),
}));
jest.mock('next/link', () => {
  return function MockLink(props: { children?: ReactNode; href?: string }) {
    const { children, href } = props;
    return <a href={href}>{children}</a>;
  };
});

// Mock auth client
jest.mock('@/lib/auth-client', () => ({
  signUp: {
    email: jest.fn(),
  },
}));

// Mock components
jest.mock('@/components/common/Toast', () => ({
  showToast: jest.fn(),
}));

// Mock constants
jest.mock('@/constants', () => ({
  AUTH_MESSAGES: {
    SIGN_UP: {
      title: 'Create Your Account',
      alreadyMember: 'Already have an account?',
      signInLink: 'Sign In',
      privacyText: 'By signing up, you agree to our',
      privacyLink: 'Privacy Policy',
    },
    SOCIAL: {
      googleSignIn: 'Sign In',
      githubSignIn: 'Sign In',
      googleSignUp: 'Sign Up',
      githubSignUp: 'Sign Up',
    },
  },
  ROUTES: {
    HOME: '/',
    SIGN_IN: '/signin',
    PRIVACY: '/privacy',
  },
  TOAST_MESSAGES: {
    SIGN_UP: {
      ERROR: {
        title: 'Sign Up Failed',
        description: 'Please try again.',
      },
    },
  },
}));

// Mock utils
jest.mock('@/utils/auth', () => ({
  extractBreadcrumbs: () => [
    { label: 'Home', href: '/', isActive: false },
    { label: 'Sign Up', href: '/signup', isActive: true },
  ],
  handleSocialAuth: jest.fn(),
}));

describe('SignUpPageContent', () => {
  it('renders signup page with form', () => {
    render(<SignUpPageContent />);

    expect(screen.getByText('Create Your Account')).toBeInTheDocument();
    expect(screen.getByText('SaaS')).toBeInTheDocument();
    expect(screen.getByPlaceholderText('Full Name')).toBeInTheDocument();
  });

  it('renders breadcrumbs', () => {
    render(<SignUpPageContent />);

    expect(screen.getByText('Home')).toBeInTheDocument();
    expect(screen.getAllByText('Sign Up').length).toBeGreaterThan(0);
  });

  it('renders footer links', () => {
    render(<SignUpPageContent />);

    expect(screen.getByText('Already have an account?')).toBeInTheDocument();
    expect(screen.getByText('Sign In')).toBeInTheDocument();
    expect(screen.getByText('Privacy Policy')).toBeInTheDocument();
  });
});
