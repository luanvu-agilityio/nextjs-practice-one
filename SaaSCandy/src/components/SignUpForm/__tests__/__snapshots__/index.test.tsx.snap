// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`SignUpForm - Snapshot matches snapshot 1`] = `
<div>
  <div
    class="flex flex-col items-center max-w-[1296px] mx-auto"
  >
    <div
      class="pb-12"
    >
      <h1
        aria-level="1"
        class="text-5xl font-semibold"
        role="heading"
      >
        Sign Up
      </h1>
      <nav
        aria-label="breadcrumb"
        class="flex items-center gap-1.5 text-sm text-slate-600 justify-center mt-4"
      >
        <ol
          class="flex flex-wrap items-center gap-1.5 break-words text-sm text-muted-foreground sm:gap-2.5"
        >
          <li
            class="inline-flex items-center gap-1.5 transition-colors hover:text-slate-900"
          >
            <a
              class="transition-colors hover:text-foreground"
              href="/"
            >
              Home
            </a>
          </li>
          <li
            aria-hidden="true"
            class="flex items-center text-slate-400"
            role="presentation"
          >
            <svg
              aria-hidden="true"
              class="lucide lucide-chevron-right h-4 w-4"
              fill="none"
              height="24"
              stroke="currentColor"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="2"
              viewBox="0 0 24 24"
              width="24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="m9 18 6-6-6-6"
              />
            </svg>
          </li>
          <li
            class="inline-flex items-center gap-1.5 transition-colors hover:text-slate-900"
          >
            <span
              aria-current="page"
              aria-disabled="true"
              class="font-normal text-foreground"
              role="link"
            >
              Sign Up
            </span>
          </li>
        </ol>
      </nav>
    </div>
    <div
      class="py-14 px-16 w-159 rounded-4xl border border-form-border-color shadow-form"
    >
      <div
        class="flex flex-col gap-10"
      >
        <div
          class="flex justify-center"
        >
          <a
            href="/"
          >
            <svg
              class="w-10 h-10"
              fill="none"
              height="32"
              viewBox="0 0 32 32"
              width="32"
              xmlns="http://www.w3.org/2000/svg"
            >
              <circle
                cx="15.9999"
                cy="15.9999"
                fill="#FFD245"
                r="15.9999"
              />
              <mask
                height="32"
                id="mask0_2053_384"
                maskUnits="userSpaceOnUse"
                style="mask-type: alpha;"
                width="32"
                x="0"
                y="0"
              >
                <circle
                  cx="15.9999"
                  cy="15.9999"
                  fill="#FFD245"
                  r="15.9999"
                />
              </mask>
              <g
                mask="url(#mask0_2053_384)"
              >
                <circle
                  cx="19.3045"
                  cy="9.53418"
                  fill="#FFE486"
                  r="15.9999"
                />
                <circle
                  cx="28.3808"
                  cy="9.53418"
                  fill="#EE7B11"
                  r="15.9999"
                />
              </g>
              <circle
                cx="24.7011"
                cy="24.7013"
                fill="#1642B9"
                r="6.29869"
                stroke="white"
                stroke-width="2"
              />
            </svg>
            <span
              class="text-3xl font-secondary text-primary"
            >
              SaaS
              <span
                class="font-medium"
              >
                Candy
              </span>
            </span>
          </a>
        </div>
        <div
          class="flex flex-col gap-8"
        >
          <div
            class="flex flex-row gap-4"
          >
            <button
              aria-label="Sign in with google"
              class="flex items-center justify-center gap-3 w-full px-4 py-3 rounded-xl text-md font-medium transition-colors border border-gray-300 bg-white text-gray-700 hover:bg-gray-50 hover:border-gray-400 focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-offset-2"
              data-provider="google"
            >
              <svg
                class="w-5 h-5"
                fill="none"
                height="24"
                viewBox="0 0 24 24"
                width="24"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M21.8055 10.0415H21V10H12V14H17.6515C16.827 16.3285 14.6115 18 12 18C8.6865 18 6 15.3135 6 12C6 8.6865 8.6865 6 12 6C13.5295 6 14.921 6.577 15.9805 7.5195L18.809 4.691C17.023 3.0265 14.634 2 12 2C6.4775 2 2 6.4775 2 12C2 17.5225 6.4775 22 12 22C17.5225 22 22 17.5225 22 12C22 11.3295 21.931 10.675 21.8055 10.0415Z"
                  fill="#FFC107"
                  stroke="#FFC107"
                />
                <path
                  d="M3.15234 7.3455L6.43784 9.755C7.32684 7.554 9.47984 6 11.9993 6C13.5288 6 14.9203 6.577 15.9798 7.5195L18.8083 4.691C17.0223 3.0265 14.6333 2 11.9993 2C8.15834 2 4.82734 4.1685 3.15234 7.3455Z"
                  fill="#FF3D00"
                  stroke="#FF3D00"
                />
                <path
                  d="M12.0002 22C14.5832 22 16.9302 21.0115 18.7047 19.404L15.6097 16.785C14.5721 17.5745 13.3039 18.0014 12.0002 18C9.39916 18 7.19066 16.3415 6.35866 14.027L3.09766 16.5395C4.75266 19.778 8.11366 22 12.0002 22Z"
                  fill="#4CAF50"
                  stroke="#4CAF50"
                />
                <path
                  d="M21.8055 10.0415H21V10H12V14H17.6515C17.2571 15.1082 16.5467 16.0766 15.608 16.7855L15.6095 16.7845L18.7045 19.4035C18.4855 19.6025 22 17 22 12C22 11.3295 21.931 10.675 21.8055 10.0415Z"
                  fill="#1976D2"
                  stroke="#1976D2"
                />
              </svg>
              Sign up with Google
            </button>
            <button
              aria-label="Sign in with github"
              class="flex items-center justify-center gap-3 w-full px-4 py-3 rounded-xl text-md font-medium transition-colors border border-gray-300 bg-white text-gray-700 hover:bg-gray-50 hover:border-gray-400 focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-offset-2"
              data-provider="github"
            >
              <svg
                class="w-5 h-5"
                fill="none"
                height="24"
                viewBox="0 0 24 24"
                width="24"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M12 2C10.6868 2 9.38642 2.25866 8.17317 2.7612C6.95991 3.26375 5.85752 4.00035 4.92893 4.92893C3.05357 6.8043 2 9.34783 2 12C2 16.42 4.87 20.17 8.84 21.5C9.34 21.58 9.5 21.27 9.5 21V19.31C6.73 19.91 6.14 17.97 6.14 17.97C5.68 16.81 5.03 16.5 5.03 16.5C4.12 15.88 5.1 15.9 5.1 15.9C6.1 15.97 6.63 16.93 6.63 16.93C7.5 18.45 8.97 18 9.54 17.76C9.63 17.11 9.89 16.67 10.17 16.42C7.95 16.17 5.62 15.31 5.62 11.5C5.62 10.39 6 9.5 6.65 8.79C6.55 8.54 6.2 7.5 6.75 6.15C6.75 6.15 7.59 5.88 9.5 7.17C10.29 6.95 11.15 6.84 12 6.84C12.85 6.84 13.71 6.95 14.5 7.17C16.41 5.88 17.25 6.15 17.25 6.15C17.8 7.5 17.45 8.54 17.35 8.79C18 9.5 18.38 10.39 18.38 11.5C18.38 15.32 16.04 16.16 13.81 16.41C14.17 16.72 14.5 17.33 14.5 18.26V21C14.5 21.27 14.66 21.59 15.17 21.5C19.14 20.16 22 16.42 22 12C22 10.6868 21.7413 9.38642 21.2388 8.17316C20.7363 6.95991 19.9997 5.85752 19.0711 4.92893C18.1425 4.00035 17.0401 3.26375 15.8268 2.7612C14.6136 2.25866 13.3132 2 12 2Z"
                  fill="black"
                />
              </svg>
              Sign up with GitHub
            </button>
          </div>
          <div
            class="relative flex items-center justify-center text-gray-background"
          >
            <div
              class="absolute inset-0 flex items-center"
            >
              <div
                class="w-full border-t border-gray-200"
              />
            </div>
            <div
              class="relative bg-white px-4"
            >
              <span
                class="text-sm text-gray-500 font-medium"
              >
                or
              </span>
            </div>
          </div>
          <form
            class="space-y-4"
            novalidate=""
          >
            <div
              class="flex flex-col gap-2 w-full"
            >
              <label
                class="text-sm font-medium text-placeholder sr-only"
                for="input-«r0»"
              >
                Name
              </label>
              <div
                class="flex items-center h-11 w-full min-w-0 rounded-lg border text-md transition-colors bg-transparent px-4 py-2.5 focus-within:border-white border-gray-background"
              >
                <input
                  aria-invalid="false"
                  aria-label="Name"
                  autocomplete="name"
                  class="w-full bg-transparent outline-none placeholder:text-placeholder disabled:pointer-events-none disabled:cursor-not-allowed"
                  id="input-«r0»"
                  name="name"
                  placeholder="Name"
                  required=""
                  type="text"
                  value=""
                />
              </div>
            </div>
            <div
              class="flex flex-col gap-2 w-full"
            >
              <label
                class="text-sm font-medium text-placeholder sr-only"
                for="input-«r1»"
              >
                Email
              </label>
              <div
                class="flex items-center h-11 w-full min-w-0 rounded-lg border text-md transition-colors bg-transparent px-4 py-2.5 focus-within:border-white border-gray-background"
              >
                <input
                  aria-invalid="false"
                  aria-label="Email"
                  autocomplete="email"
                  class="w-full bg-transparent outline-none placeholder:text-placeholder disabled:pointer-events-none disabled:cursor-not-allowed"
                  id="input-«r1»"
                  name="email"
                  placeholder="Email"
                  required=""
                  type="email"
                  value=""
                />
              </div>
            </div>
            <div
              class="flex flex-col gap-2 w-full"
            >
              <label
                class="text-sm font-medium text-placeholder sr-only"
                for="input-«r2»"
              >
                Password
              </label>
              <div
                class="flex items-center h-11 w-full min-w-0 rounded-lg border text-md transition-colors bg-transparent px-4 py-2.5 focus-within:border-white border-gray-background"
              >
                <input
                  aria-invalid="false"
                  aria-label="Password"
                  autocomplete="new-password"
                  class="w-full bg-transparent outline-none placeholder:text-placeholder disabled:pointer-events-none disabled:cursor-not-allowed"
                  id="input-«r2»"
                  name="password"
                  placeholder="Password"
                  required=""
                  type="password"
                  value=""
                />
                <span
                  class="flex ml-2 text-gray-foreground"
                >
                  <button
                    aria-label="Show password"
                    class="p-1 text-gray-foreground hover:text-gray-700 transition-colors"
                    type="button"
                  >
                    <svg
                      aria-hidden="true"
                      class="lucide lucide-eye h-4 w-4"
                      fill="none"
                      height="24"
                      stroke="currentColor"
                      stroke-linecap="round"
                      stroke-linejoin="round"
                      stroke-width="2"
                      viewBox="0 0 24 24"
                      width="24"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <path
                        d="M2.062 12.348a1 1 0 0 1 0-.696 10.75 10.75 0 0 1 19.876 0 1 1 0 0 1 0 .696 10.75 10.75 0 0 1-19.876 0"
                      />
                      <circle
                        cx="12"
                        cy="12"
                        r="3"
                      />
                    </svg>
                  </button>
                </span>
              </div>
            </div>
            <button
              class="inline-flex items-center justify-center font-semibold text-md leading-none tracking-normal rounded-xl transition-colors duration-200 focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-offset-2 disabled:pointer-events-none disabled:opacity-50 disabled:cursor-not-allowed cursor-pointer bg-orange-background text-white hover:opacity-90 focus-visible:ring-primary px-9 py-5 w-full"
              type="submit"
            >
              Sign Up
            </button>
          </form>
          <div
            class="text-center space-y-3"
          >
            <div
              class="text-md font-regular text-primary"
            >
              By creating an account you agree with our
               
              <a
                href="/privacy"
              >
                Privacy Policy
              </a>
            </div>
            <div
              class="text-md font-regular text-primary"
            >
              Already have an account?
               
              <a
                href="/signin"
              >
                Sign In
              </a>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
